lib/kitchen/verifier/pester.rb

```diff
@@ -369,42 +442,118 @@ def prepare_helpers
-          debug("Copying #{src} to #{dest}")
-          FileUtils.mkdir_p(File.dirname(dest))
-          FileUtils.cp(src, dest, preserve: true)
+      # returns the path of the modules subfolder
+      # in the sandbox, where PS Modules and folders will be copied to.
+      #
+      # @api private
+      def sandbox_module_path
+        File.join(sandbox_path, "modules")
+      end
+
+      # copy files into the 'modules' folder of the sandbox,
+      # so that copied folders can be discovered with the updated $Env:PSModulePath.
+      #
+      # @api private
+      def prepare_copy_folders
+        return if config[:copy_folders].nil?
+
+        info("Preparing to copy specified folders to #{sandbox_module_path}.")
+        kitchen_root_path = config[:kitchen_root]
+        config[:copy_folders].each do |folder|
+          debug("copying #{folder}")
+          folder_to_copy = File.join(kitchen_root_path, folder)
+          copy_if_dir_exists(folder_to_copy, sandbox_module_path)
         end
       end
 
-      def prepare_powershell_module(name)
-        FileUtils.mkdir_p(File.join(sandbox_path, "modules/#{name}"))
-        FileUtils.cp(
-          File.join(File.dirname(__FILE__), "../../support/powershell/#{name}/#{name}.psm1"),
-          File.join(sandbox_path, "modules/#{name}/#{name}.psm1"),
-          preserve: true
-        )
+      # returns an array of string
+      # Creates a flat list of files contained in a folder.
+      # This is useful when trying to debug what has been copied to
+      # the sandbox.
+      #
+      # @return [Array<String>] array of files in a folder
+      # @api private
+      def list_files(path)
+        base_directory_content = Dir.glob(File.join(path, "*"))
+        nested_directory_content = Dir.glob(File.join(path, "*/**/*"))
+        all_directory_content = [base_directory_content, nested_directory_content].flatten
+
+        all_directory_content
+      end
+
+      # Copies all test suite files into the suites directory in the sandbox.
+      #
+      # @api private
+      def prepare_pester_tests
+        info("Preparing to copy files from  '#{suite_test_folder}' to the SUT.")
+        sandboxed_suites_path = File.join(sandbox_path, "suites")
+        copy_if_dir_exists(suite_test_folder, sandboxed_suites_path)
       end
 
-      def prepare_powershell_modules
-        info("Preparing to copy supporting powershell modules.")
-        %w{PesterUtil}.each do |module_name|
-          prepare_powershell_module module_name
+      def prepare_supporting_psmodules
+        debug("Preparing to copy files from '#{support_psmodule_folder}' to the SUT.")
+        sandbox_module_path = File.join(sandbox_path, "modules")
+        copy_if_dir_exists(support_psmodule_folder, sandbox_module_path)
+      end
+
+      # Copies a folder recursively preserving its layers,
+      # mostly used to copy to the sandbox.
+      #
+      # @api private
+      def copy_if_dir_exists(src_to_validate, destination)
```

Comment:
I'd recommend renaming this to `copy_if_src_exists` as the name alone doesn't make it clear if it's validating the source or destination directories for existence.

And/or expand on the comment decorating it.